python/test/t_TensorProductExperiment_std.py

```diff
@@ -0,0 +1,255 @@
+        # Experiment 2 : Normal * 3 with 2, 2 and 1 nodes.
+        marginalSizes2 = [2, 2, 1]
+        dimension2 = len(marginalSizes2)
+        distribution2 = ot.ComposedDistribution([ot.Normal()] * dimension2)
+        experiment2 = ot.GaussProductExperiment(distribution2, marginalSizes2)
+        # Tensor product
+        collection = [experiment1, experiment2]
+        multivariateExperiment = ot.TensorProductExperiment(collection)
+        nodes, weights = multivariateExperiment.generateWithWeights()
+        # Sort
+        nodes, weights = sortNodesAndWeights(nodes, weights)
+        assert nodes.getDimension() == 5
+        assert nodes.getSize() == 24
+        assert weights.getSize() == 24
+
+        nodesExact = [
+            [-0.77459, -0.57735, -1.0, -1.0, 0.0,],
+            [-0.77459, -0.57735, -1.0, 1.0, 0.0,],
+            [-0.77459, -0.57735, 1.0, -1.0, 0.0,],
+            [-0.77459, -0.57735, 1.0, 1.0, 0.0,],
+            [-0.77459, 0.57735, -1.0, -1.0, 0.0,],
+            [-0.77459, 0.57735, -1.0, 1.0, 0.0,],
+            [-0.77459, 0.57735, 1.0, -1.0, 0.0,],
+            [-0.77459, 0.57735, 1.0, 1.0, 0.0,],
+            [0.0, -0.57735, -1.0, -1.0, 0.0,],
+            [0.0, -0.57735, -1.0, 1.0, 0.0,],
+            [0.0, -0.57735, 1.0, -1.0, 0.0,],
+            [0.0, -0.57735, 1.0, 1.0, 0.0,],
+            [0.0, 0.57735, -1.0, -1.0, 0.0,],
+            [0.0, 0.57735, -1.0, 1.0, 0.0,],
+            [0.0, 0.57735, 1.0, -1.0, 0.0,],
+            [0.0, 0.57735, 1.0, 1.0, 0.0,],
+            [0.77459, -0.57735, -1.0, -1.0, 0.0,],
+            [0.77459, -0.57735, -1.0, 1.0, 0.0,],
+            [0.77459, -0.57735, 1.0, -1.0, 0.0,],
+            [0.77459, -0.57735, 1.0, 1.0, 0.0,],
+            [0.77459, 0.57735, -1.0, -1.0, 0.0,],
+            [0.77459, 0.57735, -1.0, 1.0, 0.0,],
+            [0.77459, 0.57735, 1.0, -1.0, 0.0,],
+            [0.77459, 0.57735, 1.0, 1.0, 0.0,],
+        ]
+        weightsExact = [
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0555556,
+            0.0555556,
+            0.0555556,
+            0.0555556,
+            0.0555556,
+            0.0555556,
+            0.0555556,
+            0.0555556,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+            0.0347222,
+        ]
+        rtol = 0.0
+        atol = 1.e-5
+        ott.assert_almost_equal(nodes, nodesExact, rtol, atol)
+        ott.assert_almost_equal(weights, weightsExact, rtol, atol)
+
+    # Testing
+    testTensorProductExperiment4()
+    testTensorProductExperiment1()
+    testTensorProductExperiment2()
+    testTensorProductExperiment3()
+except:
+    import sys
+
+    print("t_GaussProductExperiment_std.py", sys.exc_info()[0], sys.exc_info()[1])
```

Comment:
the outermost try/except is useless here

Reply:
Ok. Fixed.